connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/DistributedHerderTest.java

```diff
@@ -1538,6 +1506,11 @@ public void testDoRestartConnectorAndTasksOnlyConnector() {
         Capture<Callback<TargetState>>  stateCallback = newCapture();
         worker.startConnector(EasyMock.eq(CONN1), EasyMock.anyObject(), EasyMock.anyObject(),
                 EasyMock.eq(herder), EasyMock.anyObject(TargetState.class), capture(stateCallback));
+        PowerMock.expectLastCall().andAnswer(() -> {
+            stateCallback.getValue().onCompletion(null, TargetState.STARTED);
+            return true;
+        });
+        expectQueueHerderOperation();
```

Comment:
This took a few minutes to grok. A comment might help:

```suggestion
        // Once the Connector is restarted, we should request task configurations from it, which requires waking up the herder tick thread
        expectQueueHerderOperation();
```